nav2_amcl/src/amcl_node.cpp

```diff
@@ -1403,6 +1429,10 @@ AmclNode::initPubSub()
     "amcl_pose",
     rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable());
 
+  amcl_lost_flag_pub_ = create_publisher<std_msgs::msg::Bool>(
+    "amcl_lost_flag",
+    rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable());
```

Comment:
bit of a nitpick but what do you think about giving this a positive name, aligned with the boolean status.
So rather than the negative `lost = true`, we'd publish on a topic something like `amcl_stable` or `amcl_cov_within_range`, and `true` would then indicate that all is good with amcl based localisation

Reply:
or, looking at https://github.com/ros-planning/navigation/pull/807/files, even publishing `diagnostic_msgs::DiagnosticStatus::WARN` and `diagnostic_msgs::DiagnosticStatus::OK`, if we anticipate a `std_error_level_?` in the future

Reply:
> bit of a nitpick but what do you think about giving this a positive name, aligned with the boolean status. So rather than the negative `lost = true`, we'd publish on a topic something like `amcl_stable` or `amcl_cov_within_range`, and `true` would then indicate that all is good with amcl based localisation

Good point! Thanks Aaron, always happy to discuss naming conventions! :)
How about switching to `isSomething` (or `canSomething`) — it's commonly used "good practice" naming of `Bool`-type functions.
In our case it'd be: 
`amcl_lost_flag_pub_` → `is_robot_lost_pub_`
`amcl_lost_flag` → `is_robot_lost`
 
What do you think? 

> or, looking at https://github.com/ros-planning/navigation/pull/807/files, even publishing `diagnostic_msgs::DiagnosticStatus::WARN` and `diagnostic_msgs::DiagnosticStatus::OK`, if we anticipate a `std_error_level_?` in the future
Yess, it's likely going to migrate to diagnostics in the future! 
Or bool becomes a part of advanced localization_msg type, depending on out needs. Yet, let me make changes about naming

Reply:
> or, looking at https://github.com/ros-planning/navigation/pull/807/files, even publishing `diagnostic_msgs::DiagnosticStatus::WARN` and `diagnostic_msgs::DiagnosticStatus::OK`, if we anticipate a `std_error_level_?` in the future

Yess, it's likely going to migrate to diagnostics in the future! 
Or bool becomes a part of advanced localization_msg type, depending on out needs. Yet, let me make changes about naming

Reply:
it's commonly used "good practice" naming of `Bool`-type functions. In our case it'd be: `amcl_lost_flag_pub_` → `is_robot_lost_pub_` `amcl_lost_flag` → `is_robot_lost`
> 
> What do you think?

yeah, naming is hard) im not sure my suggestions are an improvement in naming.  My suggestion was more about not using "lost", because its a negative, then setting its status as "true", which is a positive.
It would possibly lead to double negative expressions like `if (!amcl_lost_flag) { ...`, whereas I think we'd want a "consistently positive" `if (amcl_cov_within_bounds) {...`
I think the referenced PR avoids the issue by setting a enum of `OK`, `WARN` etc, so would end up something quite expressive like `if (amcl_status == DisagnosticStatus::OK) {...`

And if someone is echoing the topic during debugging, they have to keep in mind that `true` means `bad thing happened`, vs the perhaps less psychologically demanding of `true` meaning that all is as expected. 
But im not overly attached to it, just a point for discussion.

Reply:
> It would possibly lead to double negative expressions like if (!amcl_lost_flag) { ..., whereas I think we'd want a "consistently positive" if (amcl_cov_within_bounds) {...

Yet, `if (!amcl_lost_flag)` ­— is not a double-negative,
This is double-negative `if (!not_found){}`

Additionally, to improve readability, it is sometimes reasonable to bring up another `bool-type` value, e.g.: 
``` bool isRobotFeelingGood = !is_robot_lost && !is_robot_stuck ```

What I'm saying it's very likely, that our `isLost` going to become a part of some custom-made robot's state message _or,_ as you already mentioned, moves into Diagnostics :)

Reply:
If you feel like it does affect code readability, could you help with naming suggestion in our case?
I'm having non-native brain-lock :D

Reply:
> If you feel like it does affect code readability, could you help with naming suggestion in our case? I'm having non-native brain-lock :D

no, that was the hard part)

hmm, what do we imagine the consumer of this information will look like?

Reply:
I suppose it goes to BT modules,  e.g.  Small angle rotation Recovery Behavior, maybe Progress Checker, Docking Plugin

Reply:
Switching to diagnostic updater, @cmraaron